### PR TITLE
added _MapperDMPNN class and suitable tests

### PR DESCRIPTION
**Pull request by ARY2260**
# Pull Request Template
This PR adds `_MapperDMPNN` class and suitable unit tests.

### What are we doing here?
============================

This commit creates a `_MapperDMPNN` class and adds suitable unit tests.
It also modifies some type-annotations.

### Why are we doing this?
============================

The `_MapperDMPNN` class will act as a helper class to generate concatenated features and required bond-to-incoming bonds mapping, for the upcoming `DMPNNFeaturizer` class.

RDKit mol, atom, and bond type annotations are updated to the custom type annotations for Deepchem.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/feat/tests/test_mapper_class_dmpnn.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    assert len(mapper.f_ini_atoms_bonds_zero_padded) == 5
+    assert mapper.atom_to_incoming_bonds == [[0, 0], [2, 0], [1, 4], [3, 0]]
+    assert mapper.bond_to_ini_atom == [0, 1, 2, 2, 3]
+    assert mapper.b2revb == [0, 2, 1, 4, 3]
+    assert (mapper.mapping == np.asarray([[0, 0], [0, 0], [0, 4], [1, 0],
+                                          [0, 0]])).all()
+
+  def test_mapper_ring(self):
+    """
+    Test 'C1=CC=CN=C1' in _MapperDMPNN (benezene ring)
```
Comment (ARY2260) [inline deepchem/feat/tests/test_mapper_class_dmpnn.py:88]: remove 'N'

Comment (rbharath) [inline deepchem/feat/tests/test_mapper_class_dmpnn.py:88]: Typo, this should be benzene without "N"

File: deepchem/feat/tests/test_mapper_class_dmpnn.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  def get_f_atoms_zero_padded(self, mol, atom_fdim):
+    """
+    Helper method to find atoms features list (zero-padded) from the given RDKit mol
+    """
+    # get atom features
+    f_atoms = np.asarray([atom_features(atom) for atom in mol.GetAtoms()],
+                         dtype=float)
+
+    # mapping from atom index to atom features | initial input is a zero padding
+    f_atoms_zero_padded = np.asarray([[0] * atom_fdim], dtype=float)
```
Comment (ARY2260) [inline deepchem/feat/tests/test_mapper_class_dmpnn.py:39]: np.zeros

Comment (ARY2260) [reply deepchem/feat/tests/test_mapper_class_dmpnn.py]: resolved the issue

File: deepchem/feat/tests/test_mapper_class_dmpnn.py (hunk 1 above)
Comment (ARY2260) [reply deepchem/feat/tests/test_mapper_class_dmpnn.py]: resolved the typo

**Pull request merged**